### PR TITLE
URL Proxy autocompletion toolbox

### PR DESCRIPTION
**Pull request by brunobord**
refs #55
- [x] Search with Simple URL, same query argument, no data transformation,
- [x] selected items correctly loaded with the Autocomplete.selected() method
- [x] same query argument, a little data transformation,
- [x] different query argument,
- [x] support for other arguments (pagination?)
- [x] authentication,
  - [x] via headers
  - [x] via query args
  - [x] ~~correctly handling authentication errors in `core.py` -- `http_call`~~ postponed, see #60
- [x] support GET/POST methods (+ demo)

### CODE REVIEW

**Comment by brunobord** [comment]
@amessinger I'd like to have your input on how to handle an eventual error when calling the 3rd party HTTP API. at the moment, we're just raising an HTTP exception that's trapped on agnocomplete end and transmitted to the front. So, if there's any HTTP problem (wrong authentication, rate limit has been reached, HTTP 500 error on this end, whatever) the AJAX response will be thrown at the front-end the same way.

Basically, the User will search for something, and will have a blank list, because the AJAX call won't succeed.

I'm thinking about a more "user-friendly" way to handle this status. Maybe it's possible to work on a format that the front-end could proceed, a specific payload with the appropriate elements to give feedback to the user about their search, the fact that it failed, and why did it fail.
Any input?

**Comment by brunobord** [comment]
@amessinger thinking out loud: I think we need to address the error handling globally, rather than "just" on this URL Proxy toolbox, since I think that we may have to send error messages with every kind of Autocomplete class and allow the integrator to override it the easy way.
Opening a new issue right now.

**Comment by amessinger** [comment]
:tada:
